### DISCUSSION TITLE
Есть идея и реализация

### DISCUSSION THREAD

**Discussion by NanoProto** [vvip-68/LedPanelWiFi#44] [Идеи и предложения]
Привет :) 
Есть идеи, готов внести свой небольшой вклад в развитие проекта.
1) READ_SERIAL - справка по системе и работа напрямую через терминал, но надо ли оно в проекте?
2) Микрофон для сопровождение эффектов, пока занимает 2 ячейки, строки 241-242 (PIN и режимы работы) надо бы для каждого эффекта, пока времени нет... в коде резерв в 4 байта (406 - 409 на каждый эффект) думаю как раз на такой случай :)
3) Радио пульт управления... странно.... да... такой проект, такие возможности и управляется одной кнопкой...
Пульт у меня 17 клавиш ( тестировал на пультах 2-4 кнопочных, оказалось мало кнопок, 17 в самый раз :) ), пока работает через жестко прописанный коды. Код кнопки 4 байта, занимает много места в памяти... есть идея разбить по 2 байта, на маску по первым 2 байтам и остатку, получится по 2 байта на каждую кнопку... около 50 байт всего... это строки 350-400 которые к этому моменту начинают вами заполняться))

Нужны мои пять копеек?

**Comment by vvip-68** [comment]
1. Про READ_SERIAL - ничего не понятно. И зачем нужно управление напрямую через терминал тоже непонятно. Если разъясните и как это использовать (в каких сценариях) - можно подумать.

2. Тоже как-то не осознаю зачем гирлянде нужен микрофон...  Да, 4 зарезервированных байта на каждый эффект - для будущего развития. В частности планировалось привязать к эффектам звуковое сопровождение. Мне это тоже не нужно, го если кто-то делает лампу - почему бы и нет. В других прошивках вроде есть. Кто этим пользуется не знаю, но сделать в дальних планах было - почему бы и нет.

3. Пульт управления. В принципе - вещь, наверное, полезная. Руки у меня до этого не доходили. Но почему бы и нет. По памяти - в случае необходимости можно же настройки эффектов сдвинуть. Правда после обновления перенастраивать всем придется... 
Там вон дырка достаточных размеров 182-239 есть...
И с вопросами тестирования - это ж нужно будет кому-то все тестировать... И вопрошающим отвечать...
Опять же - схемы для wiki дорисовывать...

И не забываем, что всё это нужно будет сопровождать. Писать в Wiki, менять схемы подключения. И все такое. Это тоже требует времени. Кто предполагается этим будет заниматься? К примеру ESP32 - хорошо там пинов много. А с 8266 - там уже практически все занять и потребуется описывать разные комбинации что вместо чего включать и чтобы не конфликтовало.

А не хотите грамотно MQTT организовать? Оно было в предыдущей версии, но было идеологически неправильно реализовано. Для интеграции с умными домами - возникали сложности и всякие извращения. Я его использовал для удаленного управления через приложение. Приложение было нарисовано в AppInventor. Поддерживать его уже не было никакой возможности - и я перешел на управление через WEB. Приложения нет и писать некому. (Не хотите этим заняться?). Поэтому MQTT был выброшен за ненадобностью.

А вообще  тут все это обсуждать как-то неудобно по формату. Стукнитесь ко мне в телеграмм, что ли?
Хотите полноценно включиться в проект?

**Comment by NanoProto** [comment]
[LedPanelWiFi_v1.14.1218.zip](https://github.com/vvip-68/LedPanelWiFi/files/13713588/LedPanelWiFi_v1.14.1218.zip)
Не желаете взглянуть? там немного коментов оставил, слегка подправил, ну так сказать на мой взгляд. В память пока не вносил сами коды кнопок, мне конечно и чисто в программном коде все устраивает, с железными флажками так-сказать... 
По включению в проект... на вряд ли моих знаний хватит ))  но чем смогу-помогу, а может и на оборот)
Мне сам проект крайне интересен, ради нескольких функций и эффектов собрал метровую лампу и часики на подходе, могу фото скинуть, если интересно и поделиться своей реализацией.
По поводу, пинов... ну что я могу сказать, плата ESP32 стоит пару сотен... да и каждый сам может решить какие именно ему функции нужны, а поставить пару флажков и указать пины... я честно говоря не считаю чем-то запредельным для рядового пользователя решившего взять в руки паяльник и умеющего прошивать платы :) 
По поводу контактов, как с вами связаться? мне телега вполне подходит по вашему нику искать?

**Reply by vvip-68** [reply]
Спасибо, гляну что там.
И тем не менее народ все еще покупает 8266 вместо того, чтобы взять ESP32
@vvip_68

**Comment by NanoProto** [comment]
![remote](https://github.com/vvip-68/LedPanelWiFi/assets/15789357/a0a1b52b-5df0-4290-9036-7d2a50407ad0)

https://t.me/GarryLab

**Reply by vvip-68** [reply]
А как это выглядит на экране телефона?

**Reply by NanoProto** [reply]
![1716384614360](https://github.com/vvip-68/LedPanelWiFi/assets/15789357/ae8e4376-4e69-4008-84b9-17baac7cde45)
![1716384614357](https://github.com/vvip-68/LedPanelWiFi/assets/15789357/34852e5b-afa5-4c8f-a8e0-8d7fecb199f8)
![1716385083292](https://github.com/vvip-68/LedPanelWiFi/assets/15789357/1f32a9fa-dc26-4b7e-b561-a9b93915c01f)
